Sources/JWTProvider/Request+JWT.swift

```diff
@@ -36,6 +36,16 @@ extension Request {
             throw AuthenticationError.invalidBearerAuthorization
         }
 
+        // Check if JWT Token can be parsed
+        let separator: Byte = .period
+        let segments = bearer.string.components(
+            separatedBy: [separator].makeString()
+        )
+
+        guard segments.count == 3 else {
+            throw AuthenticationError.invalidCredentials
+        }
+
```

Comment:
I'd prefer catching the error from JWT. The reason being that the segment count is an implementation detail that is already handled by JWT.

```swift
do {
    // Parse the bearer string into a JWT
    return try JWT(token: bearer.string)
} catch JWTError.incorrectNumberOfSegments {
    throw AuthenticationError.invalidCredentials
}
```